src/main/java/com/hubspot/jinjava/lib/filter/AbstractFilter.java

```diff
@@ -0,0 +1,225 @@
+import com.hubspot.jinjava.interpret.InvalidInputException;
+import com.hubspot.jinjava.interpret.JinjavaInterpreter;
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import org.apache.commons.lang3.BooleanUtils;
+import org.apache.commons.lang3.NotImplementedException;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.math.NumberUtils;
+
+/***
+ * Filter base that uses Filter Jinjavadoc to construct named argument parameters.
+ * Only filters that specify name, type and defaults correctly should use this as a base
+ *
+ * @see JinjavaDoc
+ * @see JinjavaParam
+ */
+public abstract class AbstractFilter implements Filter {
+  private final Map<String, JinjavaParam> NAMED_ARGUMENTS;
+
+  public AbstractFilter() {
+    NAMED_ARGUMENTS = initNamedArguments();
+  }
+
+  abstract Object filter(
+    Object var,
+    JinjavaInterpreter interpreter,
+    Map<String, Object> parsedArgs
+  );
+
+  public Object filter(Object var, JinjavaInterpreter interpreter, String... args) {
+    throw new NotImplementedException("Not implemented");
+  }
+
+  public Object filter(
+    Object var,
+    JinjavaInterpreter interpreter,
+    Object[] args,
+    Map<String, Object> kwargs
+  ) {
+    Map<String, Object> namedArgs = new HashMap<>();
+    //Set defaults
+    NAMED_ARGUMENTS.forEach(
+      (k, v) -> {
+        if (StringUtils.isNotEmpty(v.defaultValue())) {
+          namedArgs.put(k, v.defaultValue());
+        }
+      }
+    );
+    //Process named params
+    for (Map.Entry<String, Object> passedNamedArgEntry : kwargs.entrySet()) {
+      String argName = passedNamedArgEntry.getKey();
+      Object argValue = passedNamedArgEntry.getValue();
+      int argPosition = getNamedArgumentPosition(argName);
+      if (argPosition == -1) {
+        throw new InvalidInputException(
+          interpreter,
+          "INVALID_ARG_NAME",
+          String.format(
+            "Argument named '%s' is invalid for filter %s",
+            argName,
+            getName()
+          )
+        );
+      }
+      namedArgs.put(argName, argValue);
+    }
+
+    //Process indexed params, as declared
+    for (int i = 0; i < args.length; i++) {
+      Object arg = args[i];
+      String argName = getIndexedArgumentName(i);
+      if (argName == null) {
+        throw new InvalidInputException(
+          interpreter,
+          "INVALID_ARG_NAME",
+          String.format("Argument at index '%s' is invalid for filter %s", i, getName())
+        );
+      }
+      namedArgs.put(argName, arg);
+    }
+
+    //Parse args based on their declared types
+    Map<String, Object> parsedArgs = new HashMap<>();
+    namedArgs.forEach(
+      (k, v) -> parsedArgs.put(k, parseArg(interpreter, NAMED_ARGUMENTS.get(k), v))
+    );
```

Comment:
Similar comment as above.

Reply:
These are runtime arguments so can't be initialized at init